compendium/ConfigurationAdmin/src/CMAsyncWorkService.cpp

```diff
@@ -83,20 +86,51 @@ namespace cppmicroservices
             {
                 if (threadpool)
                 {
-                    using Sig = void();
-                    using Result = boost::asio::async_result<decltype(task), Sig>;
-                    using Handler = typename Result::completion_handler_type;
-
-                    Handler handler(std::forward<decltype(task)>(task));
+                    boost::asio::post(threadpool->get_executor(), std::move(task));
+                }
+            }
 
-                    boost::asio::post(threadpool->get_executor(),
-                                      [handler = std::move(handler)]() mutable { handler(); });
```

Comment:
Ah - these lines are key - instead of posting the `task` directly, you need to wrap it up into an object which doesn't have `get_future()` or the `boost::asio` internals will call that `get_future` to try and return the future from the call to `post`. I think you can still do away with the `using` declarations and use something like:
```
boost::asio::post(threadpool->get_executor(),
                                      [task= std::move(task)]() mutable { task(); });
```